src/Couchbase.Lite.Shared/API/Database/Database.cs

```diff
@@ -745,8 +749,13 @@ public void Purge(Document document)
         public void Purge(string docId)
         {
             CBDebug.MustNotBeNull(Log.To.Database, Tag, nameof(docId), docId);
-            var document = GetDocument(docId);
-            Purge(document);
+            try {
+                LiteCoreBridge.Check(err => Native.c4db_beginTransaction(_c4db, err));
+                PurgeDocById(docId);
+                LiteCoreBridge.Check(e => Native.c4db_endTransaction(_c4db, true, e));
+            } catch (Exception) {
```

Comment:
This is not the best way to do this.  Use try / finally like other places and move the end_transaction to "finally" so that the exception is thrown up to the caller.